### PR TITLE
[HDFS exporter] Main parameters are configurable through command line

### LINKED ISSUE #58: [HDFS exporter] Values should be configurable through command line
Should be configurable: MAX_TMP_FILE_OPEN_RETRIES, MESSAGES_BEFORE_EXPIRING_WRITERS, WRITERS_EXPIRATION_DELAY, EXPIRER_PERIOD,  HEARTBEAT_PERIOD, TMP_FILE_OPEN_RETRY_PERIOD (all on top of HdfsExporter.java)

### PR DESCRIPTION
**Pull request by gray-matter**
* Using -D options
* Usage updated to show parameters name, description and default value

Closes #58

### CODE REVIEW

File: readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java [hunk 1]
```diff
@@ -153,6 +198,20 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
         heartbeat.stop().join();
     }
 
+    private static void setupProperties() {
+        maxTmpFileOpenRetries = Integer.valueOf(System.getProperty("maxTmpFileOpenRetries", "20"));
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java]: Use the associated constant MAX_TMP_FILE_OPEN_RETRIES

File: readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java [hunk 2]
```diff
@@ -217,6 +293,21 @@ private static void printHelp() {
         System.out.println("Usage:");
         System.out.println("\tjava com.criteo.hadoop.garmadon.parquet.HdfsExporter " +
                 "kafka_connection_string kafka_group temp_dir final_dir");
+
+        System.out.println();
+        System.out.println("Properties settable via -D:");
+
+        for (String parameter: Arrays.asList(MESSAGES_BEFORE_EXPIRING_WRITERS, WRITERS_EXPIRATION_DELAY, EXPIRER_PERIOD,
```
Comment (jpbempel) [inline readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java]: Maybe you can declare the array along with constants
minor cosmetic note: in a foreach the ':' is detached from any words => `for (String parameter : strArray)`

File: readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gray-matter) [reply readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java]: This got lost somewhere, thanks for pointing this out

File: readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java (hunk 2 above)
Comment (gray-matter) [reply readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java]: Done

**Pull request merged**